src/Orchard.Cms.Web/Modules/Orchard.OpenId/Controllers/AccessController.cs

```diff
@@ -192,16 +211,30 @@ public class AccessController : Controller
                 });
             }
 
-            if (application.SkipConsent)
+            var openIdSettings = await _openIdService.GetOpenIdSettingsAsync();
+            var isAuthorizationCodeFlow = (request.IsAuthorizationCodeFlow() && openIdSettings.AllowAuthorizationCodeFlow && application != null && application.AllowAuthorizationCodeFlow);
+            var isHybridFlow = (request.IsHybridFlow() && openIdSettings.AllowHybridFlow && application != null && application.AllowHybridFlow);
+            var isImplicitFlow = (request.IsImplicitFlow() && openIdSettings.AllowImplicitFlow && application != null && application.AllowImplicitFlow);
+
+            if (isAuthorizationCodeFlow || isHybridFlow || isImplicitFlow)
```

Comment:
This check is unnecessary: OpenIddict doesn't allow implementing your own response_type, so the flow is necessarily one of those 3 options.